### PR TITLE
Enable golangci lint for project

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolong.ran <rxl@apache.org>

### Modifications

Enable golangci lint for project, checks as follows:

```
    - bodyclose
    - deadcode
    - gocritic
    - goimports
    - golint
    - gosimple
    - govet
    - ineffassign
    - interfacer
    - misspell
    - staticcheck
    - structcheck
    - stylecheck
    - typecheck
    - unconvert
    - unparam
    - unused
    - varcheck
    - lll
    - prealloc
    - maligned
```

### CODE REVIEW

File: run-ci.sh [hunk 1]
```diff
@@ -29,6 +29,10 @@ go mod download
 go build ./pulsar
 go build -o pulsar-perf ./perf
 
+# Project code style check
+wget -O - -q https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s v1.18.0
```
Comment (merlimat) [inline run-ci.sh]: This should go in the Docker image so that we can cache it

**Comment by wolfstudy** [comment]
- `if` block ends with a `return` statement, so drop this `else` and outdent its block (golint)

```
if err != nil {
		return nil, err
	} else {
		return internal.NewBufferWrapper(uncompressed), nil
	}
```

- comment on exported var `FlagProfile` should be of the form `FlagProfile ...` (golint)

```
// global flags
var FlagProfile bool
```

File: run-ci.sh (hunk 1 above)
Comment (wolfstudy) [reply run-ci.sh]: Good idea @merlimat, has been fixed, please take a look again, thanks.

**Comment by wolfstudy** [comment]
@merlimat PTAL thanks

**Pull request merged**